solidity/contracts/liquidity-protection/LiquidityProtection.sol

```diff
@@ -1040,6 +1045,17 @@ contract LiquidityProtection is TokenHandler, ContractRegistryClient, Reentrancy
       * @param _reserveToken    reserve token
     */
     function reserveTokenAverageRate(IDSToken _poolToken, IERC20Token _reserveToken) internal view returns (Fraction memory) {
+        (, , uint256 averageRateN, uint256 averageRateD) = reserveTokenRates(_poolToken, _reserveToken);
```

Comment:
this will increase gas costs

Reply:
How so? The other two variables (spot rate) are used internally in order to verify the average rate's maximum permitted deviation. This is exactly why I've split that function into 2 or 3 functions.

Reply:
the cost of `reserveTokenAverageRate` is increased since we call `reserveTokenRates` that gets both versions of the rates and we only use one

Reply:
It uses both, please see my comment above.

Reply:
the reason why this isn't an issue is that when we only want one of them (spot OR average - just like in this specific case), we anyway need to get both in order to verify the max deviation.

Reply:
`reserveTokenRates` is called once from the `protectLiquidity` flow (via `reserveTokenAverageRate`), and once from the `removeLiquidity` flow (via `packRates`).